Feature/cast mocked runtime attributes (#9561)
PrefectHQ/prefect: docs/concepts/runtime-context.md

    $ python -c 'from prefect.runtime import task_run; print(task_run.fake_key)' # "foo"
    ```
    </div>

>     If the environment variable mock an existing runtime attribute, the value is casted to the same type. It only works for runtime attributes of basic types (`bool`, `int`, `float` and `str`) and `pendulum.DateTime`, for those of type `list` or `dict` they can be mocked using [monkeypatch](https://docs.pytest.org/en/latest/how-to/monkeypatch.html) or a similar tool.

Reviewer: ```suggestion
    If the environment variable mocks an existing runtime attribute, the value is cast to the same type. This works for runtime attributes of basic types (`bool`, `int`, `float` and `str`) and `pendulum.DateTime`. For complex types like `list` or `dict`, we suggest mocking them using [monkeypatch](https://docs.pytest.org/en/latest/how-to/monkeypatch.html) or a similar tool.
```
